### PR TITLE
Endermen take damage in the rain

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: src/Mobs/Enderman.cpp [hunk 1]
```diff
@@ -194,14 +194,37 @@ void cEnderman::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
...
 	}
+}
+
+
 
+
+
+bool cEnderman::CheckRain(void)
+{
+	if (GetWorld()->IsWeatherRain())
```
Comment (NiLSPACE) [inline src/Mobs/Enderman.cpp]: I'd prefer an early bail-out:
```cpp
if (!GetWorld->IsWeatherRain())
{
    return false;
}
```

File: src/Mobs/Enderman.cpp [hunk 2]
```diff
@@ -194,14 +194,37 @@ void cEnderman::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
...
+	{
+		Vector3d coords = GetPosition();
+		for (int Y = static_cast<int>(coords.y); Y < cChunkDef::Height; ++Y)
+		{
+			BLOCKTYPE Block = m_World->GetBlock(static_cast<int>(coords.x), Y, static_cast<int>(coords.z));
+			if (Block != E_BLOCK_AIR)
```
Comment (NiLSPACE) [inline src/Mobs/Enderman.cpp]: Only air or any non-solid block?

Comment (Bond-009) [inline src/Mobs/Enderman.cpp]: I tested this in vanilla and they don't take damage from rain when they are standing under a fence or another non-solid block.

**Comment by Seadragon91** [comment]
Circleci failed again :(. I see this errror on circleci: `Your build was not run - reason code (:github-failure)`. Any ideas @madmaxoft

**Comment by madmaxoft** [comment]
I see CircleCI success, did it change in the meantime?

**Comment by Bond-009** [comment]
*Your tests passed on CircleCI!* I see this, and the lastest change was just a rebase.

**Comment by Seadragon91** [comment]
I restarted it sorry, forgot to mention it.

**Pull request merged**